readthedocs/proxito/views/mixins.py

```diff
@@ -79,3 +84,47 @@ def _serve_401(self, request, project):
         res.status_code = 401
         log.debug('Unauthorized access to %s documentation', project.slug)
         return res
+
+
+class ServeRedirectMixin:
+
+    def get_redirect(self, project, lang_slug, version_slug, filename, full_path):
+        """
+        Check for a redirect for this project that matches ``full_path``.
+
+        :returns: the path to redirect the request and its status code
+        :rtype: tuple
+        """
+        redirect_path, http_status = project.redirects.get_redirect_path_with_status(
+            language=lang_slug,
+            version_slug=version_slug,
```

Comment:
I'm not quite sure why we need the `lang_slug` and `version_slug` here. It feels a bit weird. I looked through the code and they're used a couple places, but they don't seem necessary. I wonder if we can just remove it and simplify the redirects code accordingly.

Reply:
I didn't take a deep look at this on purpose. I didn't want to refactor all this without being sure that this is the direction we want to move forward but besides, because we are changing the whole logic of redirects and I felt it was too much. I'm opening an issue to track this refactor.